src/System.Net.Http/src/System/Net/Http/Managed/HttpSystemProxy.cs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
+
+namespace System.Net.Http
+{
+    internal sealed class HttpSystemProxy : IWebProxy
+    {
+        private readonly Uri _proxyUri;         // String URI for HTTPS requests
+        private string[] _bypass = null;        // list of domains not to proxy
+        private bool _bypassLocal = false;      // we should bypass domain considered local
+        private ICredentials _credentials=null;
+        private readonly WinInetProxyHelper _proxyHelper = null;
+        private readonly SafeWinHttpHandle _sessionHandle;
+
+        public static HttpSystemProxy TryCreate()
+        {
+            // This will get basic proxy setting from system using existing
+            // WinInetProxyHelper functions. If no proxy is enabled, it will return null.
+            SafeWinHttpHandle sessionHandle = null;
+
+            WinInetProxyHelper proxyHelper = new WinInetProxyHelper();
+            if (!proxyHelper.ManualSettingsOnly && !proxyHelper.AutoSettingsUsed)
+            {
+                return null;
+            }
+            if (proxyHelper.AutoSettingsUsed)
+            {
+                sessionHandle = Interop.WinHttp.WinHttpOpen(
```

Comment:
Who closes this handle and when?  If I remember correctly, the current consumption of this type is creating one per handler when the handler is created.  Presumably then this implementation should also be IDisposable, and the handler when it's disposed should query for IDisposable and call Dispose if it is?  Then Dispose would do appropriate cleanup, like closing this handle.

Reply:
The handle should be closed.  But it will be since it a SafeWinHttpHandle which is a SafeHandle() type.  The question is should there be an explicit dispose on this outer class via IDisposable pattern?

Reply:
I see; I didn't notice it was a SafeHandle.  In that case it would be eventually closed, but I do think we should make the proxy disposable.